e2e/src/helper.ts

```diff
@@ -353,3 +372,44 @@ export function ensureDirectoryExists(dir: string): void {
     fs.mkdirSync(dir);
   }
 }
+
+export function waitForManifestParsed(projectFolderName: string): Promise<void> {
+  console.log(`waitForManifestParsed '${normalizePath(projectFolderName)}'`);
+  return getLanguageServerReadyDeferred(projectFolderName).promise;
+}
+
+export async function initializeExtension(): Promise<void> {
+  const ext = extensions.getExtension('Fivetran.dbt-language-server');
+  if (!ext) {
+    throw new Error('Fivetran.dbt-language-server not found');
+  }
+  extensionApi = (await ext.activate()) as ExtensionApi;
+
+  extensionApi.manifestParsedEventEmitter.on(LS_MANIFEST_PARSED_EVENT, (languageServerRootPath: string) => {
+    console.log(`Language Server '${normalizePath(languageServerRootPath)}' ready`);
+    getLanguageServerReadyDeferred(languageServerRootPath).resolve();
+  });
+}
+
+function getLanguageServerReadyDeferred(rootPath: string): DeferredResult<void> {
+  const normalizedPath = normalizePath(rootPath);
+
+  let lsReadyDeferred = languageServerReady.get(normalizedPath);
+  if (lsReadyDeferred === undefined) {
+    lsReadyDeferred = deferred<void>();
+    languageServerReady.set(normalizedPath, lsReadyDeferred);
+  }
+
+  return lsReadyDeferred;
+}
+
+function normalizePath(rawPath: string): string {
+  return process.platform === 'win32' ? trimPath(path.normalize(rawPath)).toLocaleLowerCase() : path.normalize(rawPath);
```

Comment:
Can you explain what we solve here? Why before everything worked without such manipulations?

Reply:
Paths on Windows could have different register. E.g. 'D:/...' and 'd:/...'.